cache/disk/disk.go

```diff
@@ -944,6 +945,9 @@ func (c *diskCache) get(ctx context.Context, kind cache.EntryKind, hash string,
 	if r == nil {
 		return nil, -1, nil
 	}
+	if foundSize > c.maxProxyBlobSize {
```

Comment:
If we know the size ahead of time, we might be able to avoid the `c.Proxy.Get` call above. Maybe add a check in `availableOrTryProxy` ?